common:
    point_filter_num: 4
    image_filter_num: 1
    time_sync_en: false         # ONLY turn on when external time synchronization is really not possible
    
lidar_parameter:
    blind: 0.1
    det_range:     100.0

odometry_options:
    init_voxel_size: 0.2
    init_sample_voxel_size: 1.0
    voxel_size: 0.1 # The voxel size for the grid sampling of the new frame (before keypoints extraction) -1: don't down sampling
    sample_voxel_size: 1.5 # The size of a voxel for the selection of `keypoints` by grid sampling
    max_distance: 2000.0 # The threshold of the distance to suppress voxels from the map
    max_num_points_in_voxel: 10 # The maximum number of points per voxel of the map
    init_num_frames: 20
    min_distance_points: 0.15 # 0.15 0.01
    distance_error_threshold: 100.0 # The motion of the sensor between two frames which is considered erroneous (stops the odometry)
    motion_compensation: CONSTANT_VELOCITY # The profile of the motion compensation (IMU, CONSTANT_VELOCITY)
    initialization: INIT_CONSTANT_VELOCITY # [INIT_IMU, INIT_CONSTANT_VELOCITY]
    # motion_compensation: IMU # The profile of the motion compensation (IMU, CONSTANT_VELOCITY)
    # initialization: INIT_IMU # [INIT_IMU, INIT_CONSTANT_VELOCITY]

icp_options:
    size_voxel_map: 1.0 # 1.0 0.1 The voxel size of in the voxel map
    num_iters_icp: 5 # The number of iterations of the ICP
    min_number_neighbors: 20 # 20 5 The minimum number of neighbor points to define a valid neighborhood
    voxel_neighborhood: 1
    power_planarity: 2.0
    estimate_normal_from_neighborhood: true
    max_number_neighbors: 20
    max_dist_to_plane_icp: 0.5
    threshold_orientation_norm: 0.1 # Threshold on orientation changes (in degrees) for early termination of the ICP
    threshold_translation_norm: 0.01 # Threshold on distance changes (in m) for early termination of the ICP
    debug_print: false
    debug_viz: false
    num_closest_neighbors: 1
    threshold_voxel_occupancy: 1
    weight_neighborhood: 0.5
    weight_alpha: 0.5
    min_num_residuals: 200 # The minimum number of residuals for a valid ICP problem
    max_num_residuals: 400 # The maximum number of residuals considered (if more keypoints exist, residuals are randomly sampled)
    
map_options:
    size_voxel_map: 0.2 # The voxel size of in the voxel map for visualization
    max_num_points_in_voxel: 15 # The maximum number of points per voxel of the map for visualization
    min_distance_points: 0.01 # The minimum distance of two points for visualization
    add_point_step: 1  # The point step of adding point to color map. [default = 4]
    pub_point_minimum_views: 3  # Publish points which have been render up to "pub_pt_minimum_views" time. [default = 3]
    max_delta_trans: 0.01
    max_delta_degree: 0.03

gs:
    scale_factor: 3.0
    position_lr_init: 0.0005
    position_lr_final: 0.0005
    feature_lr: 0.001
    percent_dense: 0.01
    opacity_lr: 0.025
    scaling_lr: 0.0025
    rotation_lr: 0.0025
    lambda_dssim: 0.2
    lambda_depth_simi: 0.2
    lambda_delta_depth_simi: 0.2
    empty_gpu_cache: true
    empty_iterations: 200

gp3d:
    full_cover: false

    grid: 0.2
    min_points_num_to_gp: 10
    num_gp_side: 4
    log_time: false
    debug: false
    neighbour_size: 3

    eigen_1: 1
    max_var_mean: 0.30

    variance_sensor: 0.05
    kernel_size: 1

    image_sliding_window: 50
    curr_cam_per_iter: 1
    history_cam_per_iter: 1
